packages/babel-traverse/src/scope/index.js

```diff
@@ -740,6 +743,10 @@ export default class Scope {
     path.traverse(collectorVisitor, state);
     this.crawling = false;
 
+    if (state.unnamedDefault) {
+      state.unnamedDefault.id = this.generateUidIdentifier("default");
```

Comment:
I don't disagree that bugs will exist because of incorrect assumptions. I just don't think it's `babel-traverse` should be mutating the AST unless explicitly asked to do so. Nothing about a library called `babel-traverse` implies that it will mutate your AST automatically, without your control, as soon as you traverse.

What if people are using `babel-traverse` to statically analyze the scope in a file, but have no intention of serializing the AST back into code? Now all of a sudden they have an identifier in their AST that wasn't actually in the file, even though they haven't performed any operations that cause mutation.

What if I just want to parse the file, run it through a single Babel plugin, and they write the file back to disk? That would be impossible with this change because we'll automatically throw a new identifier in there?

What if the code is written for a real ES6 module environments and assumes the name will actually be inferred as `default`, but we're now adding `_default`?

I'm fine with having utilities to make this process easier for transforms, but this behavior should not happen automatically. Automatic transforms like the internal block-scoping and function hoisting have been a huge pain for people for this same reason, people don't expect their ASTs to be transformed when the conversion hasn't been done by an explicit plugin they loaded.